### ISSUE TITLE
[EasySecurity] Remove need for app to override the security context factory

### ISSUE THREAD

**Issue by natepage** [eonx-com/easy-monorepo#318]
Right now every app using the package MUST define it's own context factory and explicitly override the package security context factory, that sucks...

Let's find a way to avoid that.

**Comment by natepage** [comment]
Actually... After thinking about it, the apps don't even have to create a custom context, it's only if they want to.

I think we need to simplify the entire context and its factory thing.

**Comment by roman-eonx** [comment]
@natepage I thought `\EonX\EasySecurity\Bridge\Symfony\DependencyInjection\EasySecurityExtension` already registers the package SecurityContext and SecurityContextFactory already (in the `registerContext` method). Doesn't it?

**Comment by natepage** [comment]
@roman-eonx No, right now the extension is only using the SecurityContextFactory from the container to instantiate the SecurityContext and then register it as service yes but...

The factory MUST be explicitly register by the app, no default factory.
And the only way to use a custom security context, even simple, requires to create a new factory.

**Comment by roman-eonx** [comment]
@natepage then yes, I agree we should automate this in the package by registering the factory that instantiates SecurityContext. If an app needs to do something specific, it should be able to register its own implementation of the factory interface.

But please think about getting the SecurityContext instance from the container instead of simple `new SecurityContext()`. Because this way we can automatically inject some dependencies (eg. Translator).

**Comment by natepage** [comment]
@roman-eonx If you want to inject dependencies in the context that's a good use case for a custom factory.

The security context should be instantiated in this factory and not in the container otherwise no point for this factory at all.

**Comment by roman-eonx** [comment]
@natepage why not to get SecurityContext from the container in the default (package) factory implementation? Autowiring will do the rest, and the app won't need to implement a custom factory.

**Comment by natepage** [comment]
@roman-eonx Because the package purpose is to register the SecurityContext (fully configured) into the container so the app can inject it everywhere it needs and use it.

So it's an egg/chicken situation 😄

**Comment by natepage** [comment]
@roman-eonx and also if you want to inject dependencies into the security context you can also do it in a configurator so you don't have to create a factory.

**Comment by roman-eonx** [comment]
@natepage why do we need a factory then?

**Comment by natepage** [comment]
@roman-eonx in case an app needs to control the context instantiation to pass arguments to its constructor, it allows to handle all use cases without having to change the package later on.

**Comment by natepage** [comment]
@roman-eonx As discussed over a call, yes you're right, no need for a factory 😉 

Will make things simpler and also don't restrict app to customise things as they need, thank you for that.
